src/Parser/RetributionPaladin/Modules/Items/Tier20_4set.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatNumber } from 'common/format';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+class Tier20_4set extends Module {
+	static dependencies = {
+		combatants: Combatants,
+	};
+
+	holyPowerGained = 0;
+	builderId = SPELLS.BLADE_OF_JUSTICE.id;
+
+	on_byPlayer_cast(event) {
+		if(this.combatants.selected.hasBuff(SPELLS.RET_PALADIN_T20_4SET_BONUS_BUFF.id) && (event.ability.guid === SPELLS.BLADE_OF_JUSTICE.id || event.ability.guid === SPELLS.DIVINE_HAMMER_TALENT.id)){
+			//The Tier bonus turns our 2 HP builders into 3 HP builders
+			this.holyPowerGained += 1;
+		}
+		if(this.combatants.selected.hasTalent(SPELLS.DIVINE_HAMMER_TALENT.id)){
```

Comment:
This should be in the `on_initialized` event handler.

Reply:
Actually seeing as the only place this value is used is in `item()`, you can also put it in the `item()` method, making it a local variable.